doc/source/whatsnew/v0.16.1.txt

```diff
@@ -27,7 +27,7 @@ Enhancements
 API changes
 ~~~~~~~~~~~
 
-
+- When subclassing ``DataFrame``, the correct class is returned, as specified in ``_constructor_sliced`` (PR 9632)
```

Comment:
There are some related issues around. I will pick one and file it under bugfixes in `whatsnew`? You're right. It's not really an api change.